Flaky TestBlockIDsFetcherFetcher_Fetch_NoBucketIndex test

**Describe the bug**
https://github.com/cortexproject/cortex/actions/runs/8639153043/job/23684918436?pr=5782

```
--- FAIL: TestBlockIDsFetcherFetcher_Fetch_NoBucketIndex (0.00s)
    block_ids_fetcher_test.go:112: 
        	Error Trace:	/__w/cortex/cortex/pkg/storage/tsdb/bucketindex/block_ids_fetcher_test.go:112
        	Error:      	Not equal: 
        	            	expected: []ulid.ULID{ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}}
        	            	actual  : []ulid.ULID{ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, ulid.ULID{0x0, 0x0, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -5,6 +5,6 @@
        	            	  (ulid.ULID) (len=16) {
        	            	-  00000000  00 00 00 00 00 02 00 00  00 00 00 00 00 00 00 00  |................|
        	            	+  00000000  00 00 00 00 00 03 00 00  00 00 00 00 00 00 00 00  |................|
        	            	  },
        	            	  (ulid.ULID) (len=16) {
        	            	-  00000000  00 00 00 00 00 03 00 00  00 00 00 00 00 00 00 00  |................|
        	            	+  00000000  00 00 00 00 00 02 00 00  00 00 00 00 00 00 00 00  |................|
        	            	  }
        	Test:       	TestBlockIDsFetcherFetcher_Fetch_NoBucketIndex
FAIL
```

**Expected behavior**
Test should pass

--- a/pkg/storage/tsdb/bucketindex/block_ids_fetcher_test.go
+++ b/pkg/storage/tsdb/bucketindex/block_ids_fetcher_test.go
@@ -5,6 +5,7 @@ import (
 	"context"
 	"encoding/json"
 	"path"
+	"sort"
 	"sync"
 	"testing"
 	"time"
@@ -109,5 +110,8 @@ func TestBlockIDsFetcherFetcher_Fetch_NoBucketIndex(t *testing.T) {
 	require.NoError(t, err)
 	close(ch)
 	wg.Wait()
+	sort.Slice(blockIds, func(i, j int) bool {
+		return blockIds[i].Compare(blockIds[j]) < 0
+	})
 	require.Equal(t, []ulid.ULID{block1.ID, block2.ID, block3.ID}, blockIds)
 }

